src/Common.cmake

```diff
@@ -1,7 +1,17 @@
 include_guard()
 
+# This function sets ProjectOptions_SRC_DIR using the current lists path
+macro(set_project_options_src_dir)
+  get_directory_property(LISTFILE_STACK LISTFILE_STACK)
+  list(POP_BACK LISTFILE_STACK _LIST_FILE)
+  cmake_path(GET _LIST_FILE PARENT_PATH ProjectOptions_SRC_DIR)
```

Comment:
sorry, this is new in `cmake version v3.20`!

@aminya should we change min required version?

see too [cmake_minimum_required(VERSION 3.20)](https://github.com/ClausKlein/WindowsToolchain/blob/7fde47cd3dabd5a5b11b00a646a9b51d850a1a5f/Windows.Clang.toolchain.cmake#L50)

Reply:
Can you rewrite this without cmake_path? This feature is not worth the version upgrade.

Reply:
> Can you rewrite this without cmake_path? This feature is not worth the version upgrade.

but the `Toolchain` requirers this too?

Reply:
I would also like to say that 3.20 was released over a year and a half ago, on March 23rd, 2021. How conservative do we need to be? Debian stable targets 3.18, and that's about as conservative as it gets... :)

Reply:
> I would also like to say that 3.20 was released over a year and a half ago, on March 23rd, 2021. How conservative do we need to be? Debian stable targets 3.18 (July, 2020), and that's about as conservative as it gets... :)

... and with `pip install cmake ninja ` cmake v3.24 may be available.

Reply:
Yeah, I was following the Debian Stable version. That's why we are using 3.18 now. But if the Windows toolchain requires that, we can update the required version.

Reply:
Fixed in #166